test/unit/utils.test.ts

```diff
@@ -525,4 +527,32 @@ describe('driver utils', function () {
       expect(isHello(doc)).to.be.false;
     });
   });
+
+  describe('class MongoDBNamespace', () => {
+    describe('constructor()', () => {
+      it('should constrain collection property to undefined if empty string passed in', () => {
+        const namespace = new MongoDBNamespace('test', '');
+        expect(namespace).to.have.property('collection').that.is.undefined;
+      });
+    });
+
+    describe('fromString()', () => {
+      it('should constrain collection nothing follows the db name', () => {
+        const namespaceNoDot = MongoDBNamespace.fromString('test');
+        expect(namespaceNoDot).to.have.property('collection').that.is.undefined;
+        const namespaceDotFollowedByNothing = MongoDBNamespace.fromString('test.');
+        expect(namespaceDotFollowedByNothing).to.have.property('collection').that.is.undefined;
+      });
+    });
+
+    describe('withCollection()', () => {
+      it('should return new MongoDBNamespace instance', () => {
+        const dbNamespace = MongoDBNamespace.fromString('test');
+        const withCollectionNamespace = dbNamespace.withCollection('pets');
+        expect(dbNamespace).to.not.equal(withCollectionNamespace);
```

Comment:
these should be reversed, since the thing we care about testing is the result of withCollection